keylime-agent/src/main.rs

```diff
@@ -568,6 +536,11 @@ async fn main() -> Result<()> {
     {
         // Request keyblob material
         let keyblob = if config.agent.enable_iak_idevid {
+            let (Some(iak), Some(idevid), Some(attest), Some(signature)) = (iak, idevid, attest, signature)
```

Comment:
Is an `if` missing here?

```
    if let (Some(iak), Some(idevid), Some(attest), Some(signature)) = (iak, idevid, attest, signature)
```

Reply:
No I used a let-else

Reply:
Oh, I see, didn't know about this syntax. Do you know since when is this available?

My concern is if this requires too new version of the rust compiler, which might be disruptive for people stuck with older compiler version.

Reply:
Looks like it was added end of 2021. Is that too recent?

Reply:
Ok, found the answer for my own question: this is available since rust 1.65: https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html#let-else

@galmasi I remember you asked some time ago which was the oldest compiler version we supported. I think you asked if it could be version 1.67. Is this correct? If yes, I think using let-else is ok (and we should use everywhere as this looks much better then if-let-else statements)